server/status/monitor.go

```diff
@@ -0,0 +1,191 @@
+// OnRemoveRange receives RemoveRangeEvents retrieved from an storage event
+// subscription. This method is part of the implementation of
+// store.StoreEventListener.
+func (nsm *NodeStatusMonitor) OnRemoveRange(event *storage.RemoveRangeEvent) {
+	nsm.getStore(event.StoreID).removeRange(event)
+}
+
+// OnSplitRange receives SplitRangeEvents retrieved from an storage event
+// subscription. This method is part of the implementation of
+// store.StoreEventListener.
+func (nsm *NodeStatusMonitor) OnSplitRange(event *storage.SplitRangeEvent) {
+	nsm.getStore(event.StoreID).splitRange(event)
+}
+
+// OnMergeRange receives MergeRangeEvents retrieved from an storage event
+// subscription. This method is part of the implementation of
+// store.StoreEventListener.
+func (nsm *NodeStatusMonitor) OnMergeRange(event *storage.MergeRangeEvent) {
+	nsm.getStore(event.StoreID).mergeRange(event)
+}
+
+// OnStartStore receives StartStoreEvents retrieved from an storage event
+// subscription. This method is part of the implementation of
+// store.StoreEventListener.
+func (nsm *NodeStatusMonitor) OnStartStore(event *storage.StartStoreEvent) {
+	nsm.getStore(event.StoreID)
+}
+
+// OnBeginScanRanges receives BeginScanRangesEvents retrieved from an storage
+// event subscription. This method is part of the implementation of
+// store.StoreEventListener.
+func (nsm *NodeStatusMonitor) OnBeginScanRanges(event *storage.BeginScanRangesEvent) {
+	nsm.getStore(event.StoreID).beginScanRanges(event)
+}
+
+// OnEndScanRanges receives EndScanRangesEvents retrieved from an storage event
+// subscription. This method is part of the implementation of
+// store.StoreEventListener.
+func (nsm *NodeStatusMonitor) OnEndScanRanges(event *storage.EndScanRangesEvent) {
+	nsm.getStore(event.StoreID).endScanRanges(event)
+}
+
+// rangeDataAccumulator maintains a set of accumulated stats for a set of
+// ranges, computed from an incoming stream of storage events. Stats will be
+// changed by any events sent to this type; Higher level components are
+// responsible for selecting the specific ranges accumulated by a
+// rangeDataAccumulator instance.
+type rangeDataAccumulator struct {
+	stats      proto.MVCCStats
+	rangeCount int64
+	// 'scanning' is a special mode used to initialize a rangeDataAccumulator.
+	// During typical operation stats are monitored using per-operation deltas;
+	// however, when a rangeDataAccumulator is initialized it must first read
+	// the total value of all stats at the time when it is created.
+	//
+	// The scanning mode is used to facilitate this: the underlying store will
+	// initiate a scan with "beginScanRanges", and then send an AddRangeEvent
+	// for each range in the store.
+	//
+	// During a scan it is not possible for ranges to be added, removed, split
+	// or merged; however, it is possible for UpdateRangeEvents to occur during
+	// a scan. The seenScan collection is used to properly handle
+	// UpdateRangeEvents in this case.
+	isScanning bool
```

Comment:
hard to tell because it's not being used anywhere, but I pictured that this guy would just get called from a lot of different places, and probably needs to be thread safe?